### ISSUE TITLE
feat(android): allow URLs with no extensions to be used for image download

### ISSUE THREAD

**Issue by edshkliaruk** [capacitor-community/media#52]
File: MediaPlugin
Line: 331
Code fragment: `String extension = absolutePath.substring(absolutePath.lastIndexOf("."));`

Explanation.
copyFile method expects to have the absolute path with expansion, but there are cases when the path is without expansion, for example `https://domain/core/images/ec432ec1-e203-419c-948c-6b4b0146ed5f/download`.

On IOS this case works correctly, no issues.

**Comment by nkalupahana** [comment]
@edshkliaruk could you share a link that works on iOS? Having trouble ensuring this is supposed to work.

**Comment by edshkliaruk** [comment]
@nkalupahana hello,
In our case, it's endpoint which contains a downloading link of image. This problem is not existing on IOS (was tested on both platforms IOS / Android).

You can reproduce this issue creating endpoint with image (without expansion) and test it on Android.

Unfortunately, I cannot provide you a link due to NDA.

**Comment by nkalupahana** [comment]
Alright, able to reproduce with a sample server:
```js
const express = require('express')
const app = express()
const port = 3000

app.get('/', (req, res) => {
    res.send("Hello world!")
})

app.get('/image', (req, res) => {
    res.sendFile("/users/nisala/Projects/noext/IMG-4251.jpg")
})

app.listen(port, () => {
    console.log(`Listening on port ${port}`)
})
```

Unfortunately, the Android system won't recognize files without extensions as images -- they get moved to the album folder just fine, but without an extensions, the Photos/Files app won't pick them up. This would require us to do some sort of recognition on the image data. While possible, this will likely take longer to set up. Are you sure you can't just add the file extension to the end of your server URL? Really shouldn't be difficult with most servers, and it'll solve the problem on your end.

**Comment by edshkliaruk** [comment]
@nkalupahana I've already found solution but I wanted to highlight this problem for author of library to avoid breaking situation in future.

Just proposal.
Then, if Android doesn't recognise files without extensions, maybe from library side to request this URL and use this downloaded image as usual to have the similar functionality on both side (Android / IOS).

**Comment by mfschieber** [comment]
Hey @nkalupahana, I'm experiencing some issue related to this topic : while trying to download an unsplash url (https://images.unsplash.com/photo-1704072383476-edc7ad1acb84?ixid=M3w1Nzg5OHwwfDF8YWxsfDE4fHx8fHx8Mnx8MTcwNDI5NTg2MHw&ixlib=rb-4.0.3&id=xxgjZnkUM2Q), I'm having this issue : 
```
RuntimeException occurred
java.lang.RuntimeException: Copy file not found: /data/user/0/com.ipelia.yoobicv3/cache/IMG_20240104_164736162.yoobicv3/files/photo-1704072383476-edc7ad1acb84-2, error: /data/user/0/com.ipelia.yoobicv3/cache/IMG_20240104_164736162.yoobicv3/files/photo-1704072383476-edc7ad1acb84-2 (No such file or directory)
at com.getcapacitor.community.media.MediaPlugin.copyFile(MediaPlugin.java:373)
```
The issue occurs only on android, not on IOS

We tried to convert the unsplash url into base64 data so that we can download it through Media.savePhoto, but it's a very expensive calculation and the app is crashing on old android when we do that for a picture that is more than 5MB.

I checked the thread for solutions, but I can't add an extension directly to the url as it's an unsplash url :/ Maybe @edshkliaruk do you think you could you share your solution so that i can try your workaround as well ? 🙏

**Comment by edshkliaruk** [comment]
@mfschieber hey, we just accepted to download images from s3 directly with extensions, that's all. I haven't found solution for this specific case.

**Comment by mfschieber** [comment]
Hey @edshkliaruk , thank you for sharing 🙏 I don't think we can do that with unsplash urls unfortunately :/

**Comment by mfschieber** [comment]
Hey @nkalupahana, at the moment we are not able to download unsplash url, do you know if/when this feature will be implemented ? I need to know if we should search for a workaround or maybe disable the feature in our app 
thanks 🙏

**Comment by nkalupahana** [comment]
@mfschieber Sorry for the delay! I've been very busy lately, so I don't think this feature will be available anytime soon. PRs are welcome (really just need to check the mime type of the downloaded file or something, assuming that's set correctly), and I'll leave this issue open for tracking for when I do get to it.

**Comment by julienkermarec** [comment]
i have the same problem, anyone can create a PR or share a fork ?

**Comment by nkalupahana** [comment]
@julienkermarec @mfschieber @edshkliaruk fixed in v6.0.0.

**Comment by julienkermarec** [comment]
Thanks @nkalupahana , I have upgrade the plugin to v6.0.6, but i have this error :

'Cannot identify media type to save image.'

But this media is a jpg image : msfc_030823_4755_lox_dome_iv_welds_rwt_confidence_article_horizontal_still_2.jpg

![msfc_030823_4755_lox_dome_iv_welds_rwt_confidence_article_horizontal_still_2](https://github.com/capacitor-community/media/assets/3876561/41c5db4a-68cc-40b5-890b-8484254eb3e3)

Log : 
VM3:363 native Media.savePhoto (#1810531)
VM3:370 ObjectcallbackId: "1810531"methodName: "savePhoto"options: albumIdentifier: "/storage/emulated/0/Android/media/com.dfdsfsd.dsfsdf/albumName"path: "https://api.xxxx.io/api/v2/sharings?sharing_id=66684ab78e114d652f3c1dbf&workspace_uid=apollo&bearer_token=1117ecb14086eee4f506e44448456900
VM3:337 result Media.savePhoto (#1810531)
VM3:345 {message: 'Cannot identify media type to save image.', code: 'argumentError'}
